scrapy/utils/datatypes.py

```diff
@@ -69,16 +70,35 @@ class LocalCache(collections.OrderedDict):
     Older items expires first.
     """
 
-    def __init__(self, limit=None):
+    def __init__(self, limit=None, time_limit=None):
         super().__init__()
         self.limit = limit
+        self.time_limit = time_limit
 
     def __setitem__(self, key, value):
-        if self.limit:
+        now = datetime.now()
+        value = (value, now)
+
+        if not self.limit is None:
             while len(self) >= self.limit:
                 self.popitem(last=False)
+            for _, val in self.items():
+                if not time_limit is None and int((now - val[1]).total_seconds()) > self.time_limit:
```

Comment:
Why do you convert `(now - val[1]).total_seconds()` to int? You can just compare float with int. there is not any necessary to convert float to int first.

Reply:
You're correct, I deleted the cast.